filters/private/NormalProcessor.hpp

```diff
@@ -0,0 +1,72 @@
+#include <Eigen/Dense>
+
+#include <pdal/KDIndex.hpp>
+
+namespace pdal
+{
+
+using namespace Eigen;
+
+struct NormalProcessor
+{
+
+    NormalProcessor() : m_normal(Vector3d::Zero()), m_curvature(0.0)
+    {}
+
+    Vector3d m_normal;
+    double m_curvature;
+
+    void findNormal(PointView& view, PointIdList neighbors)
+    {
+        // Check if the covariance matrix is all zeros
+        auto B = math::computeCovariance(view, neighbors);
+        if (B.isZero())
+            return;
+    
+        SelfAdjointEigenSolver<Matrix3d> solver(B);
+        if (solver.info() != Success)
```

Comment:
This is utility code. You shouldn't assume that someone wants their program to crash when calling it. You have a perfectly good return code you can use or you can implement some more detailed error checking, but throwing from some utility function is yuck.